core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -692,19 +727,26 @@ public FileInfo getFileInfo(long fileId)
   public FileInfo getFileInfo(AlluxioURI path, GetStatusOptions options)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException {
     Metrics.GET_FILE_INFO_OPS.inc();
-
     try (JournalContext journalContext = createJournalContext();
-         LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.READ)) {
-      mPermissionChecker.checkPermission(Mode.Bits.READ, inodePath);
+         LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.READ);
+         MasterAuditContext auditContext = createAuditContext("getFileInfo", path, null, null)) {
+      try {
+        mPermissionChecker.checkPermission(Mode.Bits.READ, inodePath);
+      } catch (AccessControlException e) {
+        auditContext.setAllowed(false);
+        throw e;
+      }
       if (inodePath.fullPathExists()) {
         // The file already exists, so metadata does not need to be loaded.
+        auditContext.setSrcInode(inodePath.getInode());
```

Comment:
could we use a `inodePath.peekInode()` to set the src inode in the call to `createAuditContext`? it's easier to maintain if we only set the src inode in one place, and that would also handle the case of including the src inode when permission checking fails